### ISSUE TITLE
FDBCursor.update() breaks on compound keys

### ISSUE THREAD

**Issue by e45lee** [dumbmatter/fakeIndexedDB#18]
FDBCursor.update breaks if fed a compound key in certain situations.  This particular line is causing the issue:
https://github.com/dumbmatter/fakeIndexedDB/blob/master/lib/FDBCursor.js#L227

I'm guessing you want Array.equals() when given a compound key.

**Comment by dumbmatter** [comment]
Do you have an example of a key it breaks on? Better yet, a failing test. Better yet, a PR :)

**Comment by e45lee** [comment]
Here's a example for which a DataError gets thrown but shouldn't:
```
var fakeIndexedDB = require('fake-indexeddb');
var FDBKeyRange = require('fake-indexeddb/lib/FDBKeyRange');

var request = fakeIndexedDB.open('test', 3);
request.onupgradeneeded = function () {
    var db = request.result;
    var store = db.createObjectStore("books", {keyPath: ["author","isbn"]});
    store.createIndex("by_title", "title", {unique: true});

    store.put({title: "Quarry Memories", author: "Fred", isbn: 123456});
    store.put({title: "Water Buffaloes", author: "Fred", isbn: 234567});
    store.put({title: "Bedrock Nights", author: "Barney", isbn: 345678});
};
request.onsuccess = function (event) {
    var db = event.target.result;

    var tx = db.transaction("books", "readwrite");
    tx.objectStore("books").openCursor(["Fred", 123456]).onsuccess = function (event) {
      var cursor = event.target.result;
      console.log("From cursor:", cursor.value);
      var clonedValue = JSON.parse(JSON.stringify(cursor.value));
      clonedValue.price = 5.99;
      cursor.update(clonedValue);
    };

    tx.oncomplete = function () {
        console.log('All done!');
    };
};
```

I can put together a pull request that'll fix this, but I'm busy for the better part of this week, so it won't be until later.

**Comment by dumbmatter** [comment]
Thanks, you're right, that does behave differently in the browser and fakeIndexedDB. I'm also busy at the moment, but eventually one of us will fix it :)

Also, in case you don't already know... the JSON.parse and JSON.stringify calls there are not necessary. Values out of IDB are already cloned, you can freely mutate them without worrying about side effects.

**Comment by e45lee** [comment]
No problem.  Yeah, I didn't know if they were/weren't necessary as we usually don't use the IndexedDB API directly, but I thought I'd play it safe (I didn't have too much time to mess around with it this morning, as I have a talk to give tomorrow...).

Thanks!

**Comment by mstange** [comment]
I think I'm hitting this issue as well. I'm looking up an element in an index (which has a compound keyPath) using `index.openCursor(['field one', 'field two'])` and not getting any results.

Are you planning to backport the patch to v1.0?

**Comment by dumbmatter** [comment]
Wasn't planning to, but since you asked I'll give it a shot :)

**Comment by dumbmatter** [comment]
Okay, this is fixed in v1.0.12!

**Comment by mstange** [comment]
Thank you very much!

It turns out that I wasn't actually hitting this problem, I had a different bug in my own code... but everything's working now, so that's good :)
